e2e/LongHaul/service/DeviceOperations.cs

```diff
@@ -71,17 +70,23 @@ public async Task InvokeDirectMethodAsync(Logger logger, CancellationToken ct)
                                 (DateTimeOffset.UtcNow - responsePayload.CurrentTimeUtc).TotalSeconds);
                         }
 
-                        logger.Trace($"Response status: {response.Status}, payload:\n\t{JsonConvert.SerializeObject(response.PayloadAsString)}", TraceSeverity.Information);
+                        logger.Trace(
+                            $"Response status: {response.Status}, payload:\n\t{JsonConvert.SerializeObject(response.PayloadAsString)}",
+                            TraceSeverity.Information);
                         break;
                     }
-                    catch (IotHubServiceException ex) when (ex.ErrorCode == IotHubServiceErrorCode.DeviceNotOnline)
+                    catch (IotHubServiceException ex) when (ex.IsTransient is true)
```

Comment:
```suggestion
                    catch (IotHubServiceException ex) when (ex.IsTransient)
```

Reply:
If these exceptions are transient, wouldn't they be retried by the retry policy?

The case we had here was one that wasn't retriable, but we choose to retry anyway.

Reply:
Yeah, the case here is just for logging a retriable/transient exception with the "Warning" severity trace, while other exceptions would be logged as "Error" severity trace. I don't mind to remove this if we don't care about logging differnently.

Reply:
The exceptions won't be surfaced because they'll be retried. The SDK doesn't let the exception get to the user code in that case.